tests/foreman/cli/test_setting.py

```diff
@@ -36,6 +36,32 @@
 class SettingTestCase(CLITestCase):
     """Implements tests for Settings for CLI"""
 
+    @classmethod
+    def setUpClass(cls):
+        super(SettingTestCase, cls).setUpClass()
+        # Collect original defaults so we can reset them later
+        cls.default_settings = {k['name']: k['value'] for k in Settings.list()}
+
+    def setUp(self):
+        self.reset_settings = set()
+
+    def tearDown(self):
+        # Reset settings to defaults
+        for s in self.reset_settings:
+            Settings.set({'name': s, 'value': self.default_settings[s]})
+
+    def setSetting(self, name, value):
```

Comment:
just thinking if this would be thread-safe, I see there are two tests below manipulating "discovery_prefix", on the other hand, I didn't see it fail in the previous runs

Reply:
How exactly are tests multithreaded? I would hope at least that every test class or module is run within one thread and you don't have 2 threads running test from the same module.

Reply:
automation runs in 4 threads, than we have run_in_one_thread decorator that marks tests to be run sequentially, I'm not sure if tests within class are safe

https://github.com/SatelliteQE/robottelo-ci/blob/ae396713e1a809dc3a0700fc6023e71c59675c0a/scripts/satellite6-automation.sh#L119-L136

Reply:
@pgagne, we currently don't use the per-class distribution, so 2 concurrent workers can get testcases from the very same class.